feature-cancel-insurance/src/main/kotlin/com/hedvig/android/feature/cancelinsurance/data/CancelInsuranceUseCase.kt

```diff
@@ -7,22 +7,26 @@ import com.apollographql.apollo3.ApolloClient
 import com.hedvig.android.apollo.OperationResult
 import com.hedvig.android.feature.cancelinsurance.InsuranceId
 import kotlinx.coroutines.ensureActive
+import kotlinx.datetime.Clock
+import kotlinx.datetime.Instant
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toLocalDateTime
 import kotlin.coroutines.coroutineContext
+import kotlin.time.Duration.Companion.days
 
 // todo real impl
 internal class CancelInsuranceUseCase(
   private val apolloClient: ApolloClient,
 ) {
-  var failTimes = 3
-
   suspend fun invoke(
     insuranceId: InsuranceId,
     selectedDateMillis: Long,
   ): Either<OperationResult.Error, InsuranceId> {
     coroutineContext.ensureActive()
-    return if (failTimes != 0) {
-      failTimes--
-      OperationResult.Error.NetworkError("manual failure").left()
+    val submittedDate = Instant.fromEpochMilliseconds(selectedDateMillis).toLocalDateTime(TimeZone.UTC)
+    return if (submittedDate < Clock.System.now().plus(3.days).toLocalDateTime(TimeZone.UTC)) {
+      // Artificial failure when selecting in the next 3 days
+      OperationResult.Error.GeneralError("Too early").left()
```

Comment:
This is still a mock implementation, just made it fail if you click on the first 3 available days to make it easy to test both success and error cases without waiting for anything